src/dbnode/storage/index/block.go

```diff
@@ -711,27 +715,58 @@ func (b *block) aggregateWithSpan(
 			}
 
 			field, term := iter.Current()
-			batch, numAdded = b.appendFieldAndTermToBatch(batch, field, term, iterateTerms)
```

Comment:
so the problem here was that numAdded included duplicates?

Reply:
Added the writeup of the problem on the other comment, hopefully explains what was wrong